servicetalk-http-api/src/test/java/io/servicetalk/http/api/HttpResponseFactoryStatusCodeTest.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright © 2019 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.http.api;
+
+import io.servicetalk.concurrent.internal.ServiceTalkTestTimeout;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+import org.mockito.stubbing.Answer;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.withSettings;
+
+public class HttpResponseFactoryStatusCodeTest {
```

Comment:
Thanks for adding this test @volyx, it is useful. I would suggest a few changes:

- Move this to `servicetalk-http-netty` and use real factory implementations instead of mocking them. eg:

```java
final BlockingStreamingHttpResponseFactory blockingStreamingHttpResponseFactory =
        new DefaultBlockingStreamingHttpResponseFactory(INSTANCE, BufferAllocators.DEFAULT_ALLOCATOR);
```

- Instead of `HttpResponseFactory` as the reference, use `HttpResponseStatus` to validate that all statuses defined in `HttpResponseStatus` are present in all factories. You can use the public constants that are of type `HttpResponseStatus` and validate that there exists a method on every factory that can create a response with that status. Or if you have a better way to validate, feel free to implement that 😄

Reply:
Thx :) I have updated the PR